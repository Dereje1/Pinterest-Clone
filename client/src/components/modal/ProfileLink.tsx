import React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import Link from '@mui/material/Link';

interface ProfileLinkProps {
  authenticated: boolean
  userId: string
  title: string | React.ReactElement
  closePin: (_: React.SyntheticEvent, forceClose?: boolean) => void
  displayLogin: () => void
}

function ProfileLink({
  authenticated,
  userId,
  title,
  closePin,
  displayLogin,
}: ProfileLinkProps) {
  return (
    <Link
      component={RouterLink}
      underline="none"
      to={
        authenticated ? `/profile/${userId}` : ''
      }
      onMouseDown={(e) => e.preventDefault()}
      onClick={(e) => (authenticated ? closePin(e, true) : displayLogin())}
    >
      {title}
    </Link>
  );
}

export default ProfileLink;
