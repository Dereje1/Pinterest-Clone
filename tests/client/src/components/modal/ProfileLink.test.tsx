import React from 'react';
import { EnzymePropSelector, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProfileLink from '../../../../../client/src/components/modal/ProfileLink';

test('The profile link component will render for an authenticated user', () => {
  const closePin = jest.fn();
  const displayLogin = jest.fn();
  const wrapper = shallow(<ProfileLink
    title="Link title"
    userId="2"
    closePin={closePin}
    displayLogin={displayLogin}
    authenticated
  />);
  expect(toJson(wrapper)).toMatchSnapshot();
  const link: EnzymePropSelector = wrapper.find('ForwardRef(Link)');
  link.props().onMouseDown({ preventDefault: jest.fn() });
  link.props().onClick({});
  expect(link.props().to).toBe('/profile/2');
  expect(closePin).toHaveBeenCalledWith({}, true);
  expect(displayLogin).not.toHaveBeenCalled();
});

test('The profile link component will render for a non authenticated user', () => {
  const closePin = jest.fn();
  const displayLogin = jest.fn();
  const wrapper = shallow(<ProfileLink
    title="Link title"
    userId="2"
    closePin={closePin}
    displayLogin={displayLogin}
    authenticated={false}
  />);
  expect(toJson(wrapper)).toMatchSnapshot();
  const link: EnzymePropSelector = wrapper.find('ForwardRef(Link)');
  link.props().onMouseDown({ preventDefault: jest.fn() });
  link.props().onClick({});
  expect(link.props().to).toBe('');
  expect(closePin).not.toHaveBeenCalled();
  expect(displayLogin).toHaveBeenCalledWith();
});
