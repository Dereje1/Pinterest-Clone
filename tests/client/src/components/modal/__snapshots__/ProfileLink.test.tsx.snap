// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The profile link component will render for a non authenticated user 1`] = `
<ForwardRef(Link)
  component={
    {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  onClick={[Function]}
  onMouseDown={[Function]}
  to=""
  underline="none"
>
  Link title
</ForwardRef(Link)>
`;

exports[`The profile link component will render for an authenticated user 1`] = `
<ForwardRef(Link)
  component={
    {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  onClick={[Function]}
  onMouseDown={[Function]}
  to="/profile/2"
  underline="none"
>
  Link title
</ForwardRef(Link)>
`;
